dss/util/__init__.py

```diff
@@ -188,3 +190,23 @@ def _chunks():
         UploadId=mpu['UploadId'],
     )
     return parts
+
+def countdown(maximum_duration: float) -> typing.Iterator[float]:
+    """
+    Convenience generator to time limit iterated work.
+    """
+    start_time = python_time.time()
+    while True:
+        seconds_remaining = maximum_duration - (python_time.time() - start_time)
+        if 0 > seconds_remaining:
+            break
+        yield seconds_remaining
+
+def circular_generator(lst: typing.List[typing.Any]) -> typing.Any:
```

Comment:
could itertools.cycle have not been used?

Reply:
Thanks, I wasn't aware of itertools.cycle. Itertools is usually a good way to write cryptic code, but this looks like an appropriate use case.